examples/generative/molecule_generation.py

```diff
@@ -492,32 +505,34 @@ def _gradient_penalty(self, graph_real, graph_generated):
         # Compute the gradients with respect to the interpolated graphs
         grads = tape.gradient(logits, [adjacency_interp, features_interp])
         # Compute the gradient penalty
-        grads_adjacency_penalty = (1 - tf.norm(grads[0], axis=1)) ** 2
-        grads_features_penalty = (1 - tf.norm(grads[1], axis=2)) ** 2
-        return tf.reduce_mean(
-            tf.reduce_mean(grads_adjacency_penalty, axis=(-2, -1))
-            + tf.reduce_mean(grads_features_penalty, axis=(-1))
+        grads_adjacency_penalty = (1 - ops.norm(grads[0], axis=1)) ** 2
+        grads_features_penalty = (1 - ops.norm(grads[1], axis=2)) ** 2
+        return ops.mean(
+            ops.mean(grads_adjacency_penalty, axis=(-2, -1))
+            + ops.mean(grads_features_penalty, axis=(-1))
         )
 
     def inference(self, batch_size):
-        z = tf.random.normal((batch_size, LATENT_DIM))
+        z = keras.random.normal(
+            shape=(batch_size, LATENT_DIM), seed=self.seed_generator
+        )
         reconstruction_adjacency, reconstruction_features = model.decoder.predict(z)
         # obtain one-hot encoded adjacency tensor
-        adjacency = tf.argmax(reconstruction_adjacency, axis=1)
-        adjacency = tf.one_hot(adjacency, depth=BOND_DIM, axis=1)
+        adjacency = ops.argmax(reconstruction_adjacency, axis=1)
+        adjacency = ops.one_hot(adjacency, num_classes=BOND_DIM, axis=1)
         # Remove potential self-loops from adjacency
-        adjacency = tf.linalg.set_diag(adjacency, tf.zeros(tf.shape(adjacency)[:-1]))
+        adjacency = adjacency * (1.0 - ops.eye(NUM_ATOMS, dtype="float32")[None, None])
         # obtain one-hot encoded feature tensor
-        features = tf.argmax(reconstruction_features, axis=2)
-        features = tf.one_hot(features, depth=ATOM_DIM, axis=2)
+        features = ops.argmax(reconstruction_features, axis=2)
+        features = ops.one_hot(features, num_classes=ATOM_DIM, axis=2)
         return [
             graph_to_molecule([adjacency[i].numpy(), features[i].numpy()])
             for i in range(batch_size)
         ]
 
     def call(self, inputs):
         z_mean, log_var = self.encoder(inputs)
-        z = Sampling()([z_mean, log_var])
+        z = Sampling(seed=self.seed_generator)([z_mean, log_var])
```

Comment:
Do not create a layer inside `call`, instead create it in `__init__` and call it here

Reply:
Done